standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -4201,6 +4202,20 @@ public NotificationEventResponse getNextNotification(long lastEventId, int maxEv
                                                        NotificationFilter filter) throws TException {
     NotificationEventRequest rqst = new NotificationEventRequest(lastEventId);
     rqst.setMaxEvents(maxEvents);
+    return getNextNotificationsInternal(rqst, false, filter);
+  }
+
+  @Override
+  public NotificationEventResponse getNextNotification(NotificationEventRequest request,
```

Comment:
Instead of defining a new getNextNotification api, would it be cleaner if we add `allowGapsInEventIds` as an option in NotificationEventRequest?

Reply:
Actually adding this to the request object will not help since even if the server does the filtering (or throwing the exception) the client API as it is implemented currently will continue to throw the exception. We cannot change the exiting client side API since I believe there are applications (e.g hive replication) which rely on this behavior currently. Hence either ways I think the change will involve client side changes.

Reply:
That make sense. Thanks for the clarification.